flake.nix

```diff
@@ -3,14 +3,14 @@
 {
   description = "A collection of packages for the Nix package manager";
 
-  outputs = { self }:
+  inputs.lib = "path:./lib";
```

Comment:
I might be wrong about this, but this has a serious concern that the nixpkgs `flake.lock` has to be updated whenever lib is updated. And if its not, then the `nixpkgs` flake will break for everyone. I don't think thats a fair ask for lib contributors who might not be familiar with flakes or not even have it installed. 
Until theres proper tooling - checks, auto update lockfile - I don't think nixpkgs should do any sub flaking. I would say leave nixpkgs's flake.nix as it is with the `import ./lib` and just add a flake.nix for lib.

Reply:
addressed, thanks!

Reply:
Isn't the problem still there even if you have only the `inputs.lib` line? The nixpkgs flake will break whenever lib changes

Reply:
Addressed again, Thanks!